File: 2032-LargestOddNumberInString/2032-LargestOddNumberInString.py
# Last updated: 8/13/2025, 8:16:48 PM
import sys
class Solution:
    def largestOddNumber(self, num: str) -> str:
        oddIndex = -1

        for i in range(len(num)-1 , -1 , -1) :
            if (int(num[i])%2!=0) :
                return num[0:i+1]
        return ""

File: 2032-LargestOddNumberInString/2032-LargestOddNumberInString.java
// Last updated: 8/13/2025, 8:16:54 PM
class Solution {
    public String largestOddNumber(String num) {
        int l = num.length();
        int oddIndex = -1;
        String ans = "";
        for(int i = l-1 ; i>=0;i--){
            int number = num.charAt(i) - '0';
            if(number%2!=0){
                 return num.substring(0 , i+1);
            }
        }
        return "";
    }
}